src/modelfree/configs/multi/train.py

```diff
@@ -77,7 +162,85 @@ def _finetune_spec(envs=None):
     return spec
 
 
+def _ysnp_hyper_finetune_defense(train):
+    _sparse_reward(train)
+    # Checkpoints take up a lot of disk space, only save every ~500k steps
+    train['checkpoint_interval'] = 2 ** 19
+    train['total_timesteps'] = int(10e6)
+    train['env_name'] = 'multicomp/YouShallNotPassHumans-v0'
+    # "Victim" here is the adversary
+    train['victim_index'] = 0
+    train['normalize_observations'] = False
+    train['load_policy'] = {
+        'path': '1',
+        'type': 'zoo',
+    }
+    spec = {
+        'config': HYPERPARAM_SEARCH_VALUES,
+        'num_samples': 100,
+    }
+    return spec
+
+
+def _ysnp_hyper_against_finetune_defense(train):
+    _sparse_reward(train)
+    train['checkpoint_interval'] = 2 ** 19
+    train['total_timesteps'] = int(10e6)
+    train['env_name'] = 'multicomp/YouShallNotPassHumans-v0'
+    # Victim is back to being the finetuned victim again
+    train['victim_index'] = 1
+    train['victim_type'] = "ppo2"
+    # Best victim from hyper_finetune_defense, according to win rate
+    train['normalize'] = True
+    spec = {
+        'config': HYPERPARAM_SEARCH_VALUES,
+        'num_samples': 100,
+    }
+    return spec
+
+
+def _ysnp_finetune_zoo_20e6(train):
+    _sparse_reward(train)
+    # Checkpoints take up a lot of disk space, only save every ~500k steps
+    train['checkpoint_interval'] = 2 ** 19
+    train['total_timesteps'] = int(20e6)
+    train['env_name'] = 'multicomp/YouShallNotPassHumans-v0'
+    # "Victim" here is the adversary
+    train['victim_index'] = 0
+    train['normalize_observations'] = False
+    train['load_policy'] = {
+        'path': '1',
+        'type': 'zoo',
+    }
+    spec = {
+        "config": {
+            "seed": tune.grid_search([0, 1, 2, 3, 4])
+        }
+    }
+    return spec
+
+
+def _ysnp_adversary_retraining_20e6(train):
+    _sparse_reward(train)
+    train['env_name'] = 'multicomp/YouShallNotPassHumans-v0'
+    train['victim_index'] = 1
+    train['normalize'] = True
+    # Train from scratch
+    train['load_policy'] = {
```

Comment:
Is this needed? Don't we default to not loading a policy / training from scratch?

Reply:
I find it to be cleaner to keep track of and reason about what a given config is doing if I explicitly articulate that we're loading from scratch, rather than just doing waterfall defaults in the cases where those apply.